script.js

```diff
@@ -36,18 +36,23 @@ function setup() {
 	for (let i = 0; i < modes.length; i++) {
 		let mode = modes[i]
 		let button = createButton(mode.name)
+    button.attribute('name', mode.name)
 		button.size(BUTTON_WIDTH, BUTTON_HEIGHT)
 		button.style('font-size', '24px')
 		button.position(
 			(width - BUTTON_WIDTH) / 2,
       //700 was originally "height", but it messes up on smaller viewports. Not entirely sure why.
 			(700 - modes.length * (BUTTON_HEIGHT + BUTTON_GAP) * i) / 2 - BUTTON_GAP
 		)
-		button.mousePressed(() => {
-			currentMode = modes[i]
-			modeSelectButtons.forEach((b) => b.hide())
-			state = PLAY_STATE
-		})
+    button.mousePressed(() => {
+      currentMode = modes[i]
+      modeSelectButtons.forEach((b) => b.hide())
+      if (currentMode.name == 'Grade Ten+') {
+        // Initiates setup with phrases
+        currentMode.setup(phraseImporters[file_selector.selectedIndex])
```

Comment:
should actually check if any new phrase-collection has been imported before overriding the default set for the current game mode.

Reply:
I'm not sure this is entirely necessary since the selector automatically defaults to the standard set of phrases, but I can still add a check and some clarification if needed.